src/unix_term.rs

```diff
@@ -66,41 +65,71 @@ pub fn terminal_size(out: &Term) -> Option<(u16, u16)> {
     }
 }
 
-pub fn read_secure() -> io::Result<String> {
-    let mut f_tty;
-    let fd = unsafe {
-        if libc::isatty(libc::STDIN_FILENO) == 1 {
-            f_tty = None;
-            libc::STDIN_FILENO
-        } else {
-            let f = fs::OpenOptions::new()
-                .read(true)
-                .write(true)
-                .open("/dev/tty")?;
-            let fd = f.as_raw_fd();
-            f_tty = Some(BufReader::new(f));
-            fd
+enum Input<T> {
+    Stdin(io::Stdin),
+    File(T),
+}
+
+impl<T> Input<T>
```

Comment:
Can we `impl<T: BufRead> BufRead for Input<T>` instead?

Reply:
Sure. Done.